backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/CodeGenerationInfo.kt

```diff
@@ -15,3 +15,17 @@ internal fun IrClass.isNonGeneratedAnnotation(): Boolean =
                 !this.annotations.hasAnnotation(serialInfoAnnotationFqName)
 
 private val serialInfoAnnotationFqName = FqName("kotlinx.serialization.SerialInfo")
+
+/**
+ * We don't need to generate RTTI in some cases, e.g. Objective-C external classes.
+ */
+internal fun IrClass.requiresRtti(): Boolean = when {
+    // TODO: Support more cases
+    this.isExternalObjCClass() -> false
```

Comment:
What about Kotlin subclasses of Objective-C classes? Do they actually require any RTTI?
If so, this probably should be fixed.